File: Corte1/Python/Práctica5/Estadisticas.py
# Define una clase llamada Estadisticas
class Estadisticas:

    @staticmethod
    # Método estático que calcula la suma de los elementos de un arreglo
    def sum(a):
        total = 0  # Variable donde se guardará la suma total

        # Recorre cada elemento del arreglo a
        for x in a:
            total += x  # En cada interaccion se suma el valor de x al total

        return total  # Devuelve la suma total

    @staticmethod
    # Método estático que calcula el promedio de los elementos
    def average(a):
        return Estadisticas.sum(a) / len(a)  # Llama al método sum y divide entre la cantidad de elementos


# Función principal main
def main():
    grades = [80, 75, 90, 60, 95]  # Arreglo que ya tiene los datos predefinidos

    # Imprime la suma de las notas usando el método sum
    print("Sum:", Estadisticas.sum(grades))

    # Imprime el promedio de las notas usando el método average
    print("Average:", Estadisticas.average(grades))


# Verifica si el archivo se está ejecutando directamente
if __name__ == "__main__":
    main()  # Llama a la función principal

File: Corte1/C++/Estadísticas/Estadisticas.cpp
#include <iostream>   // Librería que permite usar entrada y salida de datos como cout y cin
using namespace std;  // Permite usar cout y otras funciones sin escribir std:: antes

// Define la clase llamada Estadisticas
class Estadisticas 
{
public: // Indica que los elementos dentro pueden usarse fuera de la clase

    // Función estática que calcula la suma de los elementos de un arreglo
    static int sum(const int a[], int n) 
    {
        int total = 0;// Variable donde se guardará la suma total

        // Bucle que recorre el arreglo desde la posición 0 hasta n-1
        for (int i = 0; i < n; i++) 
            total += a[i];// En cada vuelta se suma el elemento del arreglo al total

        return total;// Devuelve la suma total de los elementos
    }

    // Función estática que calcula el promedio de los elementos del arreglo
    static double average(const int a[], int n) 
    {
        // Llama a la función sum para obtener la suma y la divide entre n
        return (double)sum(a, n) / n; // Se convierte a double para obtener decimales
    }
};

// Función principal main
int main() 
{
    // Declaración de un arreglo llamado grades que ya estan dados los datos
    int grados[] = {80, 75, 90, 60, 95};

    // Calcula el número de elementos del arreglo
    int n = sizeof(grados) / sizeof(grados[0]);
    /*sizeof(grados) da el tamaño total del arreglo en bytes
      sizeof(grados[0]) da el tamaño de un elemento
      al dividirlos obtenemos la cantidad de elementos*/

    // Imprime en pantalla la suma de los elementos usando la función sum
    cout << "Sum: " << Estadisticas::sum(grados, n) << "\n";

    // Imprime en pantalla el promedio usando la función average
    cout << "Average: " << Estadisticas::average(grados, n) << "\n";

    return 0; // Indica que el programa terminó correctamente
}